torch_dicom/preprocessing/pipeline.py

```diff
@@ -176,24 +190,37 @@ def dataloader(self) -> DataLoader:
         )
 
     @staticmethod
-    def _save_as(result: Dict[str, Any], dest: Path) -> Path:
+    def _save_as(result: Dict[str, Any], dest: Path, output_format: str) -> Path:
         dicom = result["dicom"]
         assert isinstance(dicom, Dicom)
 
-        # Save the DICOM
-        dest_path = dest / "images" / f"{dicom.StudyInstanceUID}" / f"{dicom.SOPInstanceUID}.dcm"
-        dest_path.parent.mkdir(exist_ok=True, parents=True)
-        dicom.save_as(dest_path, write_like_original=False)
+        assert output_format in ("png", "dcm")
```

Comment:
We already perform the same validation in `__post_init__()` with a descriptive ValueError. I think it's fine to leave this as an assert since it's only performing a recheck of a condition in a private method. However, the assert should match against `OUTPUT_FORMATS` instead of the current tuple.